classes/class-truelayer-gateway.php

```diff
@@ -142,34 +144,40 @@ public function country_currency_check( $order = false ) {
 	 * @return array
 	 */
 	public function process_payment( $order_id ) {
+		$order       = wc_get_order( $order_id );
 		$settings    = get_option( 'woocommerce_truelayer_settings', array() );
 		$epp_enabled = $settings['truelayer_payment_page_type'] ?? 'HPP';
 
-		$response = TrueLayer()->api->create_payment( $order_id );
+		$payment = TrueLayer()->api->create_payment( $order );
```

Comment:
`create_payment` seem to expect an `$order_id` while a `WC_Order` is passed.

https://github.com/krokedil/truelayer-woocommerce/blob/main/classes/requests/post/class-truelayer-request-create-payment.php#L40

Reply:
The function we call later will allow a [order, order id, or WP_Post.](https://woocommerce.github.io/code-reference/files/woocommerce-includes-wc-order-functions.html#source-view.78
)